### ISSUE TITLE
Non-null default for requestTimeout?

### ISSUE THREAD

**Issue by dandv** [node-influx/node-influx#178]
Continuing the discussion from #171 -

I've managed to work around the timeouts associated with numerous small posts by batching requests 200 points at a time, like this:

```js
const BUFFER_SIZE = 200;
let points = [];
let dumpingInProgress = false;

// just for debugging
setInterval(function () {
  console.log(new Date().toISOString(), 'Dumping in progress:', dumpingInProgress);
}, 5000);

function dumpPoints() {
  if (dumpingInProgress) return;
  if (points.length >= BUFFER_SIZE) {
    try {
      dumpingInProgress = true;
      influx.writePoints('fooSeries', points.slice(0, BUFFER_SIZE) /* elements beyond BUFFER_SIZE go in the next buffer */, 
          {precision: 'ms'}, function (err) {
        dumpingInProgress = false;
        if (err) {
          // Just keep the points in the buffer. They'll be written eventually.
          console.error(new Date().toISOString(), `ERROR: Couldn't write ${points.length} points: ${err}`);
        } else {
          points.splice(0, BUFFER_SIZE);  // shift the buffer to the left by BUFFER_SIZE points
          console.info(new Date().toISOString(),
              `${BUFFER_SIZE} points dumped to InfluxDB. ${points.length} remaining.`
          );
        }
      });
    } catch (error) {
      console.error('Dumping error:', error);
    }
  }
}

yourevent.on(..., function (data) {
  points.push([
    {
      time: new Date(),
      [data.someKey]: data.someValue
    },
    {tag1: 'foo'}
  ]);
  dumpPoints();
});
```

This approach leads to a payload of about 16KB, occuring about every 5 seconds.

However, after ~10 minutes, no points were dumped, and the debugging interval function showed that `dumpingInProgress` was true, which means the callback was never called.

Turns out that was because `requestTimeout` was `null`. Is there any particular reason for that default?

In my scenario, I only have one host, but I don't see how never timing out helps, even when having multiple hosts. Eventually, all hosts would be kicked out of rotation.

After connecting with a timeout of 20 seconds, the same code above has been working much better for hours. The connection was made like this:

```js
const influx = require('influx')({
  hosts: ['https://mydomaim.com:12345'],
  // take a host out of balancing for 40 seconds if a request fails or takes longer than 20
  requestTimeout: 20000,
  failoverTimeout: 40000  
});
```

The host still goes down, but that's not our library's fault, and the batch write code above recovers well, and writes all backlogged data.

Since [InfluxDB boasts handling extremely high-load scenarios](https://www.influxdata.com/influxdb-is-27x-faster-vs-mongodb-for-time-series-workloads/) (27x faster than MongoDB), it seems a more sensible `requestTimeout` option would be 20-30 seconds, rather than forever.

**Comment by connor4312** [comment]
We definitely should time out be default. However, implementing it in 4.x is _technically_ a breaking change. I'll put this on the 5.0.0 roadmap (and it is, in fact, [already done](https://github.com/node-influx/node-influx/blob/5.0.0/src/pool.ts#L112))

**Comment by connor4312** [comment]
I'll go ahead and close this since the `Pool` is complete and includes timeouts as of https://github.com/node-influx/node-influx/commit/9a2e882855059b9790a1fdc9578efd16b894fdf7

**Comment by connor4312** [comment]
These changes have been published in `5.0.0-alpha.1`. You can install this via:

```
npm install --save influx@next
```
